### PR TITLE
VP-366: Export support in Catalog module related changes

### PR DESCRIPTION
**Pull request by akak1977**
ICloneable implementation for Catalog models;
Extended some SearchCriteria and SearchResults;

### CODE REVIEW

File: VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs [hunk 1]
```diff
@@ -162,5 +163,62 @@ public virtual void ReduceDetails(string responseGroup)
                 Variations = null;
             }
         }
+
+        #region ICloneable members
+        public virtual object Clone()
+        {
+            var result = MemberwiseClone() as CatalogProduct;
+
+            if (Properties != null)
```
Comment (yecli) [inline VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs]: Please use shorter syntax  in similar cases:
result.Properties = Properties?.Select(x => x.Clone() as Property) ?? result.Properties;

File: VirtoCommerce.Domain/Catalog/Model/Category.cs [hunk 2]
```diff
@@ -1,10 +1,11 @@
+using System;
 using System.Collections.Generic;
 using VirtoCommerce.Domain.Commerce.Model;
 using VirtoCommerce.Platform.Core.Common;
 
 namespace VirtoCommerce.Domain.Catalog.Model
 {
-    public class Category : AuditableEntity, ILinkSupport, ISeoSupport, IHasOutlines, IHasImages, IHasProperties
+    public class Category : AuditableEntity, ILinkSupport, ISeoSupport, IHasOutlines, IHasImages, IHasProperties, ICloneable
```
Comment (yecli) [inline VirtoCommerce.Domain/Catalog/Model/Category.cs]: Please make required objects and related ones ICloneable the same way as in this commit (it is not merged in v3 yet):
https://github.com/VirtoCommerce/vc-platform-core/pull/434/commits/7c33eb647dd6cd000ac4c92bdc0212cfe7d71e46

File: VirtoCommerce.Domain/Catalog/Model/PropertyDictionaryValue.cs [hunk 3]
```diff
@@ -28,5 +25,12 @@ public class PropertyDictionaryValue : Entity
         public string Value { get; set; }
         //The dictionary item identifier 
         public string ValueId { get; set; }
+
+        #region ICloneable members
+        public virtual object Clone()
+        {
+            return MemberwiseClone() as Property;
```
Comment (yecli) [inline VirtoCommerce.Domain/Catalog/Model/PropertyDictionaryValue.cs]: Why `as Property`?

Comment (akak1977) [reply VirtoCommerce.Domain/Catalog/Model/PropertyDictionaryValue.cs]: It's copy-pasting bug

**Pull request merged**